breadcord/helpers.py

```diff
@@ -278,12 +278,29 @@ def __init__(self, to_wrap: logging.Formatter | None = None) -> None:
         super().__init__()
         self._wrapped = to_wrap or logging.Formatter()
 
+    @staticmethod
+    def _strip_colour_ansi(string: str) -> str:
+        """Strips ANSI colour codes from a string."""
+        # If this does not match everything we want it to, I take no responsibility - Fripe
+        new = ''
+        i = 0
+        while i < len(string):
+            if string[i] == '\x1b' and string[i + 1] == '[':
+                end = string.find('m', i+2)
+                if end == -1:
+                    new += string[i:]
+                    break
+                if all(char in '0123456789;' for char in string[i+2:end]):
+                    i = end + 1
+                    # Avoid IndexError since the escape sequence could be at the end of the string
+                    if i >= len(string):
+                        break
+            new += string[i]
+            i += 1
+        return new
+
```

Comment:
Algorithm breakdown:
if we find an "ESC[" we find the next "m" (end of the ANSI code)
if no "m" exists we kill the loop and append the rest of the string, since there can't be any more ANSI codes
if the characters between the "ESC[" and "m" are all numbers or semicolons, we move the index to after the "m"
if the index is at the end of the string, we kill the loop and append the rest of the string